test/integration/xds_delegate_extension_integration_test.cc

```diff
@@ -0,0 +1,236 @@
+#include <atomic>
+
+#include "envoy/config/xds_resources_delegate.h"
+#include "envoy/service/discovery/v3/discovery.pb.h"
+#include "envoy/service/runtime/v3/rtds.pb.h"
+
+#include "test/common/grpc/grpc_client_integration.h"
+#include "test/config/v2_link_hacks.h"
+#include "test/integration/http_integration.h"
+#include "test/integration/xds_delegate_test_config.pb.h"
+#include "test/test_common/registry.h"
+#include "test/test_common/utility.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace {
+
+// A test implementation of the XdsResourcesDelegate extension.
+class TestXdsResourcesDelegate : public Config::XdsResourcesDelegate {
+public:
+  void onConfigUpdated(const std::string& /*authority_id*/,
+                       const std::string& /*resource_type_url*/,
+                       const std::vector<Config::DecodedResourceRef>& /*resources*/) override {
+    OnConfigUpdatedCount++;
+  }
+
+  std::vector<envoy::service::discovery::v3::Resource>
+  getResources(const std::string& /*authority_id*/,
+               const std::string& /*resource_type_url*/) override {
+    // TODO(abeyad): implement this and test for it when we add support for loading config from the
+    // delegate in a subsequent PR.
+    return {};
+  }
+
+  static int getOnConfigUpdatedCount() { return OnConfigUpdatedCount; }
+
+private:
+  static std::atomic<int> OnConfigUpdatedCount;
+};
+std::atomic<int> TestXdsResourcesDelegate::OnConfigUpdatedCount{0};
+
+// A factory for creating the TestXdsResourcesDelegate test implementation.
+class TestXdsResourcesDelegateFactory : public Config::XdsResourcesDelegateFactory {
+public:
+  ProtobufTypes::MessagePtr createEmptyConfigProto() override {
+    return std::make_unique<test::envoy::config::xds::TestXdsResourcesDelegateConfig>();
+  }
+
+  std::string name() const override { return "envoy.config.xds.test_delegate"; };
+
+  Config::XdsResourcesDelegatePtr createXdsResourcesDelegate(const ProtobufWkt::Any&,
+                                                             ProtobufMessage::ValidationVisitor&,
+                                                             Api::Api&) override {
+    return std::make_unique<TestXdsResourcesDelegate>();
+  }
+};
+
+std::string bootstrapConfig() {
```

Comment:
is this mostly cargo culted from somewhere else and if so can we snag it from there rather than copying it?

Reply:
So, I did lift most of this from the rtds_integration_test's bootstrap config, but I'm wondering what's the best way to share this config, if any.  My test isn't really related to rtds_integration_test, other than using RTDS for config made setting the test's bootstrap easier, so I don't think my test should depend on rtds_integration_test. Is there a good place to put "similar config strings" like this for the test infra?  Should I add a new file/library to https://github.com/envoyproxy/envoy/blob/main/test/test_common ?

Reply:
I realized what you meant with using ConfigHelper, so I updated the test to use ConfigHelper + addConfigModifier